deb-package/build

```diff
@@ -0,0 +1,52 @@
+#!/bin/sh
+#
+# Builds a Debian package for MXNET
+#
+# Copyright sociomantic labs GmbH 2017.
+# Distributed under the Apache License, Version 2.0.
+# (See accompanying LICENSE file or copy at
+# https://www.apache.org/licenses/LICENSE-2.0.txt)
+set -xeu
+
+# Set up package
+pkgname=libmxnet
+pkgversion_number=$(git describe --tags --dirty |
+                    sed 's/^v\(.*\)$/\1/' |
+                    sed 's/-\([0-9]\+\)-\(g[0-9a-f]\+\)/+\1~\2/' |
+                    sed 's/\(~g[0-9a-f]\+\)-dirty$/-dirty\1/' |
+                    sed 's/-dirty/~dirty.'`date +%Y%m%d%H%M%S`'/'
+                   )
+pkgdistro=$(lsb_release -cs)
+pkgversion="${pkgversion_number}~${pkgdistro}"
```

Comment:
Going by typical Ubuntu package versions, ~ is usually used before this kind of version qualifier.  But I'm happy to tweak it however would be preferred.

The main thing I'd like to address is to ensure that any `+socioN` in the semver tag becomes a `-socioN` in the package version.  That's presumably another straightforward `sed` command (`s/+socio/-socio/` should suffice).

Then we wind up with a package version like `0.10.0-socio1~xenial` which is comparable to (for example) the `git` package version `1:2.13.0-0ppa1~ubuntu14.04.1`.